qualtran/bloqs/data_loading/qroam_clean_test.py

```diff
@@ -0,0 +1,141 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import numpy as np
+import pytest
+import sympy
+
+from qualtran.bloqs.data_loading.qroam_clean import (
+    _qroam_clean_multi_data,
+    _qroam_clean_multi_dim,
+    QROAMClean,
+    QROAMCleanAdjoint,
+)
+from qualtran.symbolics import ceil
+
+
+def test_bloq_examples(bloq_autotester):
+    bloq_autotester(_qroam_clean_multi_data)
+    bloq_autotester(_qroam_clean_multi_dim)
+
+
+def test_t_complexity_1d_data_symbolic():
+    # 1D data, 1 dataset
+    N, b, k = sympy.symbols('N b k')
+    bloq = QROAMClean.build_from_bitsize((N,), (b,), log_block_sizes=(k,))
+    K = 2**k
+    expected_toffoli = ceil(N / K) + (K - 1) * b - 2
+    assert bloq.t_complexity().t == 4 * expected_toffoli
+    bloq_inv = bloq.adjoint()
+    assert isinstance(bloq_inv, QROAMCleanAdjoint)
+    inv_k = sympy.symbols('kinv')
+    inv_K = 2**inv_k
+    bloq_inv = bloq_inv.with_log_block_sizes(log_block_sizes=(inv_k,))
+    expected_toffoli_inv = ceil(N / 2) / inv_K + inv_K
+    assert bloq_inv.t_complexity().t == 4 * expected_toffoli_inv
```

Comment:
Maybe out of scope but we should probably make a utility function which uses the new cost framework to get Toffoli counts directly but hides some of the boilerplate.